diff -r 541c445d6c13 src/pkg/http/request.go
--- a/src/pkg/http/request.go	Fri Jun 24 16:04:17 2011 +1000
+++ b/src/pkg/http/request.go	Fri Jul 08 21:10:47 2011 -0700
@@ -151,6 +151,12 @@
 	// TLS-enabled connections before invoking a handler;
 	// otherwise it leaves the field nil.
 	TLS *tls.ConnectionState
+
+	// Env allows for associating data with a particular request,
+	// such as session state, decoded data stored in cookies, or
+	// any other data that might be passed from one layer of http
+	// request processing (middleware, if you will) to the next.
+	Env map[string]interface{}
 }
 
 // ProtoAtLeast returns whether the HTTP protocol used
